src/components/Privacy/createComponent.js

```diff
@@ -19,18 +19,25 @@ export default ({
   defaultConsent,
   consent,
   sendSetConsentRequest,
-  validateSetConsentOptions
+  validateSetConsentOptions,
+  consentHashStore,
+  doesIdentityCookieExist
 }) => {
-  const consentByPurpose = assign(
-    { [GENERAL]: defaultConsent },
-    readStoredConsent()
-  );
+  let consentByPurpose = { [GENERAL]: defaultConsent };
+
+  const identityCookieExists = doesIdentityCookieExist();
+  const consentCookieExists = readStoredConsent()[GENERAL] !== undefined;
+  if (!identityCookieExists || !consentCookieExists) {
+    consentHashStore.clear();
```

Comment:
Yes. The failing test would be this:

defaultConsent = "in"
1. setConsent to "out"
2. drop identity cookie
3. reload
4. sendEvent (sent)
5. reload
6. sendEvent

Expected: the last sendEvent should go out because the current ECID is different than the ECID used when the consent was set to "out".
Actual: the last sendEvent is queued.